File: 153.find-minimum-in-rotated-sorted-array.py
#
# @lc app=leetcode id=153 lang=python
#
# [153] Find Minimum in Rotated Sorted Array
#

# @lc code=start
class Solution(object):
    def findMin(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        l, h = 0, len(nums)-1
        while l < h:
            m = l + (h-l)//2
            if nums[m] <= nums[h]:
                h = m
            else:
                l = m+1
        return nums[l]
# @lc code=end

File: 153.find-minimum-in-rotated-sorted-array.cpp
/*
 * @lc app=leetcode id=153 lang=cpp
 *
 * [153] Find Minimum in Rotated Sorted Array
 */
#include <vector>
using namespace std;

// @lc code=start
class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0, h = nums.size() - 1, m;
        while (l < h) {
            m = l + (h - l) / 2;
            if (nums[m] <= nums[h])
                h = m;
            else
                l = m + 1;
        }
        return nums[l];
    }
};
// @lc code=end

